File: Xamarin/printQRCODE.cs
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Sockets;
using System.Text;

namespace gertecEasyPrint
{
    internal class printQRCODE
    {
        public static void PrintQRCODE(Socket socket, String qrText)
        {
            string qrCode = "";
            int len = (qrText).Length + 3;
            byte pl = (byte)(len % 256);
            byte ph = (byte)(len / 256);

            Encoding m_encoding = Encoding.GetEncoding("iso-8859-1");
            qrCode += m_encoding.GetString(new byte[] { 29, 40, 107, 4, 0, 49, 65, 50, 0 });
            qrCode += m_encoding.GetString(new byte[] { 29, 40, 107, 3, 0, 49, 67, 8 });
            qrCode += m_encoding.GetString(new byte[] { 29, 40, 107, 3, 0, 49, 69, 48 });
            qrCode += m_encoding.GetString(new byte[] { 29, 40, 107, pl, ph, 49, 80, 48 });
            qrCode += qrText;
            qrCode += m_encoding.GetString(new byte[] { 29, 40, 107, 3, 0, 49, 81, 48 });

            byte[] reset = Encoding.ASCII.GetBytes("\x0A");
            byte[] nBytes = Encoding.ASCII.GetBytes(qrCode);
            socket.Send(nBytes);
            socket.Send(reset);
        }
    }
}

File: ReactNative/printQRCODE.js
const net = require('net')

function printQRCODE(socket, text) {
    var qrCODE = '';
    var len = text.length + 3;
    var pl = len % 256;
    var ph = ~~(len / 256);

    qrCODE += String.fromCharCode.apply(null, [29, 40, 107, 4, 0, 49, 65, 50, 0]);
    qrCODE += String.fromCharCode.apply(null, [29, 40, 107, 3, 0, 49, 67, 8]);
    qrCODE += String.fromCharCode.apply(null, [29, 40, 107, 3, 0, 49, 69, 48]);
    qrCODE += String.fromCharCode.apply(null, [29, 40, 107, pl, ph, 49, 80, 48]);
    qrCODE += text;
    qrCODE += String.fromCharCode.apply(null, [29, 40, 107, 3, 0, 49, 81, 48]);
    socket.write(qrCODE);
}

export {printQRCODE};

